commcare_connect/opportunity/models.py

```diff
@@ -209,6 +183,34 @@ def display_name(self):
             return "---"
 
 
+class CompletedModule(XFormBaseModel):
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name="completed_modules",
+    )
+    module = models.ForeignKey(LearnModule, on_delete=models.PROTECT)
+    opportunity = models.ForeignKey(Opportunity, on_delete=models.PROTECT)
+    opportunity_access = models.ForeignKey(OpportunityAccess, on_delete=models.CASCADE, null=True)
```

Comment:
Only this line was added and the model was moved to be below OpportunityAccess to avoid any undefined warnings.